__tests__/index.spec.js

```diff
@@ -1,18 +1,253 @@
 /* eslint-disable no-undef */
-const clientRP = require('reportportal-client');
+const { getOptions, RPClient } = require('./mocks/reportportal-client.mock');
+const JestReportPortal = require('../index');
 
+const testItemStatuses = { PASSED: 'passed', FAILED: 'failed', SKIPPED: 'pending' };
+const GLOBAL_CONFIG = {};
+const options = getOptions();
+const currentDate = new Date();
+const RealDate = Date;
+const testResult = {
+    testResults: [
+        {
+            title: 'Title',
+            status: 'failed',
+            ancestorTitles: ['Suite name'],
+            failureMessages: 'error message',
+            invocations: 1,
+        }
+    ]
+};
 
 describe('index script', () => {
-    test('require reportportal-client package', () => {
-        const result = new clientRP({
-            debug: false,
-            token: "00000000-0000-0000-0000-000000000000",
-            endpoint: "http://your-instance.com:8080/api/v1",
-            launch: "LAUNCH_NAME",
-            project: "PROJECT_NAME"
-        });
-
-        expect(result).toBeDefined();
-        expect(result).not.toBe(null);
+    let reporter;
+
+    beforeAll(() => {
+        reporter = new JestReportPortal(GLOBAL_CONFIG, options);
+        reporter.client = new RPClient();
+    });
+
+    beforeEach(() => {
+        global.Date = jest.fn(
+            (...props) =>
+                props.length
+                    ? new RealDate(...props)
+                    : new RealDate(currentDate)
+        );
+        Object.assign(Date, RealDate);
+    });
+
+    afterEach(() => {
+        jest.clearAllMocks();
+        global.Date = RealDate;
+    });
+
+    afterAll(() => {
+        reporter = null;
```

Comment:
I think it's not necessary. There are no references outside of this module, so everything should be clear after execution.

Reply:
I faced problems with the heap limit, this code helped me with it.